modules/js/src/embindgen.py

```diff
@@ -635,7 +635,7 @@ def gen_function_binding(self, func, class_info):
                 if ptr_type in type_dict:
                     ret_type = type_dict[ptr_type]
             for key in type_dict:
-                if key in ret_type:
+                if key in ret_type and "DetectorParameters" not in ret_type:
```

Comment:
Can you please describe why DetectorParameters cannot be parsed with common code?

Reply:
As mentioned in the issue the  .replace operator is used to replace parts of the string with strings from a type_dict. During this operation the Param string from DetectorParameters is replaced with **cv::Param**, leading then to the error .constructor(select_overload<Ptr<**Detectorcv::Parameters**>()> in the binding.cpp.

Is that what you meant?

Reply:
Yeah, exactly! Do you know where exactly in code generation script there is such pattern matching? In my point of view, it's better to fix `cv::Param` replacement rather than add an excetion for `DetectorParameters`

Reply:
You are right. It's not a nice way because for future bindings this could happen again. I can have a look this afternoon. However, I have not understood the binding generation completely as the code is not that well documented and very long. I just debugged myself to the error. So if anyone with more experience with the binding generation has an idea it would be great.

Reply:
So I had a quick look at it. I don't see a quick way to make this more elegant from my side, sorry. Maybe someone else has an idea?!

Reply:
Seems I found a problem. Please use the following construction to manage types replacement more accurate:
```patch
--- a/modules/js/src/embindgen.py
+++ b/modules/js/src/embindgen.py
@@ -636,7 +636,10 @@ class JSWrapperGenerator(object):
                     ret_type = type_dict[ptr_type]
             for key in type_dict:
                 if key in ret_type:
-                    ret_type = ret_type.replace(key, type_dict[key])
+                    # Replace types. Instead of ret_type.replace we use regular
+                    # expression to exclude false matches.
+                    # See https://github.com/opencv/opencv/issues/15514
+                    ret_type = re.sub('(^|[^\w])' + key + '($|[^\w])', type_dict[key], ret_type)
```
in our case, works extra replacement from `Param` to `cv::Param` for `Ptr<DetectorParameters>`. Proposed patch applies replacement not just by pattern but pattern with non-alphanumeric symbols before and after.

Reply:
Awesome! Updated it.